services/cd-service/pkg/repository/transformer_test.go

```diff
@@ -731,6 +736,670 @@ func TestCreateApplicationVersion(t *testing.T) {
+			}
+		})
+	}
+}
+
+func TestUndeployApplicationCommitPath(t *testing.T) {
+	type TestCase struct {
+		Name                   string
+		Transformers           []Transformer
+		ExistentCommitPaths    []string
+		NonExistentCommitPaths []string
+	}
+
+	generateLargeTest1 := func(versionCount uint64) TestCase {
+		name := fmt.Sprintf("Create two applications %d times and then undeploy one of them", versionCount)
+
+		transformers := make([]Transformer, 0)
+		commitPaths1 := make([]string, 0)
+		commitPaths2 := make([]string, 0)
+
+		for i := uint64(0); i < versionCount; i++ {
+			commitID := randomCommitID()
+			transformers = append(transformers, &CreateApplicationVersion{
+				Application:    "app1",
+				SourceCommitId: commitID,
+			})
+			transformers = append(transformers, &CreateApplicationVersion{
+				Application:    "app2",
+				SourceCommitId: commitID,
+			})
+
+			commitPaths1 = append(commitPaths1, path.Join("commits", commitID[:2], commitID[2:], "applications", "app1", ".empty"))
+			commitPaths2 = append(commitPaths2, path.Join("commits", commitID[:2], commitID[2:], "applications", "app2", ".empty"))
+		}
+
+		transformers = append(transformers, &CreateUndeployApplicationVersion{
+			Application: "app1",
+		})
+		transformers = append(transformers, &UndeployApplication{
+			Application: "app1",
+		})
+
+		existentCommitPaths := commitPaths2
+		NonExistentCommitPaths := commitPaths1
+		return TestCase{
+			Name:                   name,
+			Transformers:           transformers,
+			ExistentCommitPaths:    existentCommitPaths,
+			NonExistentCommitPaths: NonExistentCommitPaths,
+		}
+
+	}
+
+	tcs := []TestCase{
+		{
+			Name: "Create one application with SHA1 commit ID and then undeploy it",
+			Transformers: []Transformer{
+				&CreateApplicationVersion{
+					Application:    "app",
+					SourceCommitId: "aaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaa",
+				},
+				&CreateUndeployApplicationVersion{
+					Application: "app",
+				},
+				&UndeployApplication{
+					Application: "app",
+				},
+			},
+			NonExistentCommitPaths: []string{
+				"commits/aa/aaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaa/applications/app/.empty",
+			},
+		},
+		{
+			Name: "Create two applications and then undeploy one of them",
+			Transformers: []Transformer{
+				&CreateApplicationVersion{
+					Application:    "app1",
+					SourceCommitId: "aaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaa",
+				},
+				&CreateApplicationVersion{
+					Application:    "app2",
+					SourceCommitId: "aaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaa",
+				},
+				&CreateUndeployApplicationVersion{
+					Application: "app1",
+				},
+				&UndeployApplication{
+					Application: "app1",
+				},
+			},
+			ExistentCommitPaths: []string{
+				"commits/aa/aaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaa/applications/app2/.empty",
+			},
+			NonExistentCommitPaths: []string{
+				"commits/aa/aaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaa/applications/app1/.empty",
+			},
+		},
+		generateLargeTest1(30),
```

Comment:
is this the latest version?
I feel like I looked at your screen and this was already changed :thinking: 
Either way please do not have a function returning an entire testcase, but just the transformers. 
The "expected output" of test should be hard coded to make it simple to read many tests at a glance.

Reply:
You're right, I refactored all large test functions except this one. Now it should be fixed.